src/org/rust/lang/structure/RustModTreeElement.kt

```diff
@@ -0,0 +1,25 @@
+package org.rust.lang.structure
+
+import com.intellij.ide.structureView.StructureViewTreeElement
+import com.intellij.ide.structureView.impl.common.PsiTreeElementBase
+import org.rust.lang.core.psi.*
+
+class RustModTreeElement(file: RustModItem) : PsiTreeElementBase<RustModItem>(file) {
+    override fun getPresentableText() = element?.name
+
+    override fun getChildrenBase(): Collection<StructureViewTreeElement> {
+        return element?.getDeclarations()?.map { toTreeElement(it) }?.filterNotNull().orEmpty()
```

Comment:
I think it's better to use `element.items` here:

* it preserves more type information
* `getDeclarations` might go away soon